cmd/ipfs/init.go

```diff
@@ -106,8 +111,16 @@ environment variable:
 			}
 		}
 
+		identity, err := config.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
+			options.Key.Size(nBitsForKeypair),
+			options.Key.Type(algorithm),
+		})
```

Comment:
As we discussed we should get an error if you try to pass bits and an algorithm that doesn't accept/need bits (e.g. Ed25519). We should probably do this in `config.CreateIdentity` instead of here though.

Note: If we're going to change this then do you think we should allow bits=256 for Ed25519 or error? If setting bits == error then the sharness tests below may need to change, but we can do that in the PR that bubbles up the config changes.

Reply:
Maybe not allowing to set the bits is better. Otherwise, we would have to require setting them (to be symmetric with RSA usage), and this would break existing users of elliptic keys.

Reply:
Are you preferring the changes to `config.CreateIdentity` to be in a separate PR?

Reply:
It has to be it's in a different repo (go-ipfs-config). However, if you want to do the change there and bubble it up in this PR that's fine too, otherwise we can have a separate bubble PR